engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetSourceLocationNode.java

```diff
@@ -0,0 +1,51 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.TruffleException;
+import com.oracle.truffle.api.TruffleStackTrace;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(
+    type = "Meta",
+    name = "get_source_location",
+    description = "Returns a textual representation of the location of the callsite.")
+public class GetSourceLocationNode extends Node {
+
+  Text execute(Object _this, long framesToSkip) {
+    var location = renderSourceLocation(framesToSkip);
+    return Text.create(location);
+  }
+
+  private static final String UNKNOWN = "Unknown Location";
+
+  @TruffleBoundary
+  private String renderSourceLocation(long framesToSkip) {
+    long ix = framesToSkip + 1;
+    if (ix > Integer.MAX_VALUE) throw new IllegalArgumentException("Too large offset.");
+    var throwable = new PanicException(Text.create("Nothing"), getRootNode());
```

Comment:
This is a hack for a thing that Truffle already exposes much more elegantly. Use https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/TruffleRuntime.html#iterateFrames-com.oracle.truffle.api.frame.FrameInstanceVisitor- instead.